datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/bool_op.rs

```diff
@@ -68,11 +70,21 @@ where
     fn update_batch(
```

Comment:
In order to realize the benefit of this blocked implementation I think you will need to change the state of the accumulators so that instead of a single large buffer

```rust
    /// values per group
    values: BooleanBufferBuilder,
```

The state is held in chunks like
```rust
    /// blocks of values per group
    values: Vec<BooleanBufferBuilder>
```

(or possibly this to support taking them out individually)
```rust
    /// blocks of values per group, None when taken
    values: Vec<Option<BooleanBufferBuilder>>
```

Reply:
Yes, I agree. 

Maybe we can modify it when we start to support the blocked mode in the specific `GroupAccumulator` or `GroupValues` impl. Seems it will be a too large change, if we change all of them once.